hal/build.gradle

```diff
@@ -12,7 +12,8 @@ ext.addHalCompilerArguments = { binary->
         }
     }
     tasks.withType(CppCompile) {
-        binary.cppCompiler.args "-DNAMESPACED_PRIORITY"
+        binary.cppCompiler.args "-DNAMESPACED_PRIORITY",
+               "-I" + project(':ni-libraries').file('include').absolutePath
```

Comment:
What files do we need from the ni stuff? I would definitely like to not need them on other platforms, so it they're just for constants, or just a single header I'd rather update the script that moves the file fully into the Hal includes.

Reply:
It's FRCComm.h; it's not just the constants, it's the whole interface. I'd definitely defer to an alternate approach.

Reply:
Well if all of the functions have fully fleshed out versions, we have a script that can just move it into the Hal fully.

Reply:
I'm not following; you you give me a pointer to the script or to another example?

Reply:
https://github.com/wpilibsuite/allwpilib/blob/master/ni-libraries/build.gradle#L95

We would just add another copy spec to copy that file from the NI-libraries into the HAL. We run that task manually whenever we update the headers we get from NI.

Reply:
Yah.  I see that, and see that it would be easy to add this.

But I have to admit I was puzzled; I didn't add this include file, I just moved it.  So I didn't create this dependency - so why didn't it fail for Athena builds?

What I learned  'feels' wrong to me; we depend on ni-libraries all over the place, and we do it with a, imho, poorly named method.

That is, for all athena builds, we already add ni-libraries includes and libraries to many (most?) of our builds; for non athena builds we silently do nothing.

If we did something like this instead:
https://gist.github.com/jwhite66/4c2a6ef1525f1923eb9dc46e09740806

it seems like it would work fine to me.  I may not fully understand the issue with UsageReporting.h, but it feels to me that would also allow us to drop that copy and put it back into ni-libraries.

I'm not going to get hung up on this, I'm happy to add to the copy and move on, but if my alternate approach looks reasonable, I'm willing to test it and work up a PR request for it instead.